flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/operations/CatalogSinkModifyOperation.java

```diff
@@ -37,20 +37,23 @@
 	private final Map<String, String> staticPartitions;
 	private final QueryOperation child;
 	private final boolean overwrite;
+	private final List<?> tableHints;
```

Comment:
why use `List<?>`?

Reply:
The module `flink-table-api-java` has no Calcite as dependency, use `List<?>` instead of `List` to fix the compiler warnings.

Reply:
I think we need to extract useful information from List<RelHint> and place it here.

Reply:
The `RelHint` itself is a POJO which has hint name and hint options as attributes, passing through these `RelHint`s and hand-over them to planner for specific manipulation can simplify the codes.

But i'm also fine if we pre-process these `RelHint`s and extract the contents we care about, such as the dynamic options or some other attributes (index in the future) directly in `CatalogSinkModifyOperation`.